pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1504,14 +1506,44 @@ func doRestorePlan(
 	}()
 
 	currentVersion := p.ExecCfg().Settings.Version.ActiveVersion(ctx)
+
+	// We support restoring a backup taken on a cluster on a specific major
+	// version into a cluster that is on the same version or the next major
+	// version.
+	minimumRestorableVersion := p.ExecCfg().Settings.Version.BinaryMinSupportedVersion()
+	if !envutil.EnvOrDefaultBool("COCKROACH_ENFORCE_MIN_RESTOREABLE_VERSION", true) {
+		// Set the minimumRestoreableVersion to a version that will be less than any
+		// backup version rendering the check a no-op.
+		//
+		// TODO(adityamaru): Delete this block once all the restore roachtest
+		// fixtures have been updated to be 22.2.0+.
+		minimumRestorableVersion = roachpb.Version{Major: 1}
+	} else if !build.IsRelease() {
+		minimumRestorableVersion, err = minimumRestorableVersion.VersionWithoutDevOffset()
+		if err != nil {
+			return err
+		}
+	}
 	for i := range mainBackupManifests {
-		if v := mainBackupManifests[i].ClusterVersion; v.Major != 0 {
-			// This is the "cluster" version that does not change between patches but
-			// rather just tracks migrations run. If the backup is more migrated than
-			// this cluster, then this cluster isn't ready to restore this backup.
-			if currentVersion.Less(v) {
-				return errors.Errorf("backup from version %s is newer than current version %s", v, currentVersion)
-			}
+		v := mainBackupManifests[i].ClusterVersion
+		if v.Major == 0 {
+			return errors.WithHint(errors.Newf("the backup is from a version older than our "+
+				"minimum restoreable version %s", minimumRestorableVersion),
+				"refer to our documentation about restoring across versions: https://www.cockroachlabs.com/docs/v22.2/restoring-backups-across-versions.html")
```

Comment:
Does the v.Major == 0 case imply we never recorded a version in the manifest?  If so, does it definitely need to happen before the currentVersion.Less check since current version will always be greater? that would help keep the two related checks next to each other.